echopype/echodata/echodata.py

```diff
@@ -242,7 +242,7 @@ def group_paths(self) -> Set[str]:
     def __get_dataset(node: DataTree) -> Optional[xr.Dataset]:
         if node.has_data or node.has_attrs:
             # validate and clean dtypes
-            return sanitize_dtypes(node.ds)
+            return sanitize_dtypes(node.to_dataset())
         return None
```

Comment:
I found it a bit curious that `__get_datset` is set as a static method. I wonder if this has anything to do with how it is used in multiple places, like `__getitem__`, `__setitem`, `__setattr__` -- so this function would get evaluated multiple times, which seems not the most efficient thing to do. Does decorating it as a static method remove the redundancy??

Reply:
@oftfrfbf: pinging you here since you may be running into these in #1420, and also this PR won't be changing the use of `__get_dataset`. Tanks!

Reply:
I may be wrong, but I'm fairly sure the static decorator is there because the function itself does not need the class to be used. It could be a standalone function, but because it's used so often in the class, it's better to just put it in the class, so I don't think it removed this redundancy.

Reply:
So yeah, I think I agree with you on that. It really only should be set on the stuff that actually sets the items, not fetches them. Fetching them for view we can send a `DatasetView` object, and for something to be actually modified or set in the `EchoData` object, `Dataset` should be sent instead.